vsintegration/src/FSharp.VS.FSI/fsiSessionToolWindow.fs

```diff
@@ -534,17 +534,16 @@ type internal FsiToolWindow() as this =
             executeTextNoHistory null text
         with _ -> ()
 
-    let executeInteraction dbgBreak dir (filename: string) topLine text =
-        // Preserving previous functionality, including the #directives...
-        let interaction =
-            "\n"
-          + (sprintf "# silentCd @\"%s\" ;; " dir) + "\n"
-          + (if dbgBreak then "# dbgbreak\n" else "")
-          + (sprintf "# %d @\"%s\" " topLine filename) + "\n"
-          + text + "\n"
-          + "# 1 \"stdin\"" + "\n" (* stdin line number reset code *)
-          + ";;" + "\n"
-
+    let executeInteraction dbgBreak dir filename topLine (text:string) =
+        let interaction = $"""
+#interactiveprompt "hide"
+#silentCd @"{dir}";;
+{if dbgBreak then "#dbgbreak" else ""}
+#{topLine} @"{filename}"
```

Comment:
In the "before" version there was a space between # and topLine

```suggestion
# {topLine} @"{filename}" 
```

Reply:
syntactically it is not needed, personally i prefer directives attached to their #.